tools/luh2/luh2.py

```diff
@@ -0,0 +1,128 @@
+
+    # Determine if we are saving a new regridder or using an old one
+    # TO DO: add check to handle if the user enters the full path
+    # TO DO: check if its possible to enter nothing with the argument
+    regrid_reuse = False
+    # If we are merging files together, we assume that the weights file
+    # being supplied exists on file
+    if (not isinstance(args.luh2_merge_file,type(None))):
+        regrid_reuse = True
+
+    # Regrid the luh2 data to the target grid
+    # TO DO: provide a check for the save argument based on the input arguments
+    regrid_luh2,regridder_luh2 = RegridConservative(ds_luh2, ds_regrid_target,
+                                                    args.regridder_weights, regrid_reuse)
+
+    # Regrid the inverted ice/water fraction data to the target grid
+    regrid_land_fraction = regridder_luh2(ds_luh2_static)
+
+    # Adjust the luh2 data by the land fraction
+    # TO DO: determine if this is necessary for the transitions and management data
+    regrid_luh2 = regrid_luh2 / regrid_land_fraction.landfrac
+
+    # Correct the state sum (checks if argument passed is state file in the function)
+    regrid_luh2 = CorrectStateSum(regrid_luh2)
+
+    # Add additional required variables for the host land model
+    # Add 'YEAR' as a variable.
+    # This is an old requirement of the HLM and should simply be a copy of the `time` dimension
+    # If we are merging, we might not need to do this, so check to see if its there already
+    if (not "YEAR" in list(regrid_luh2.variables)):
+        regrid_luh2["YEAR"] = regrid_luh2.time
+        regrid_luh2["LONGXY"] = ds_regrid_target["LONGXY"] # TO DO: double check if this is strictly necessary
+        regrid_luh2["LATIXY"] = ds_regrid_target["LATIXY"] # TO DO: double check if this is strictly necessary
+
+    # Rename the dimensions for the output.  This needs to happen after the "LONGXY/LATIXY" assignment
+    if (not 'lsmlat' in list(regrid_luh2.dims)):
+        regrid_luh2 = regrid_luh2.rename_dims({'lat':'lsmlat','lon':'lsmlon'})
+
+    # Merge existing regrided luh2 file with merge input target
+    # TO DO: check that the grid resolution 
+    # We could do this with an append during the write phase instead of the merge
+    if (not(isinstance(args.luh2_merge_file,type(None)))):
+        ds_luh2_merge = ImportData(args.luh2_merge_file,args.begin,args.end,merge_flag=True)
+        #ds_luh2_merge = ds_luh2_merge.merge(regrid_luh2)
+        regrid_luh2 = regrid_luh2.merge(ds_luh2_merge)
+
+    # Write the files
+    # TO DO: add check to handle if the user enters the full path
+    # TO DO: add check if user sets the output the same as the input
+    if (args.output == None):
+        output_filename = 'LUH2_timeseries.nc'
+    else:
+        output_filename = args.output
+
+
+    output_file = os.path.join(os.getcwd(),output_filename)
+    print("generating output: {}".format(output_file))
+    regrid_luh2.to_netcdf(output_file)
+
+def CommandLineArgs():
```

Comment:
In the add_argument calls, I mentioned the `default` option as possibly helpful.

I showed an example in this file (from the ctsm5.2 branch):
/glade/work/slevis/git/mksurfdata_toolchain/tools/mksurfdata_esmf/gen_mksurfdata_namelist.py

Reply:
resolved via 8cbb7ab204c2f2eb397d2cfec1da8f3471506da6